app/src/main/java/org/p2p/wallet/send/model/SendSolanaFee.kt

```diff
@@ -135,6 +139,13 @@ data class SendSolanaFee constructor(
         val totalNeeded = feeRelayerFee.totalInSpl + inputAmount
         val isEnoughSolBalance = solToken?.let { !it.totalInLamports.isLessThan(feeRelayerFee.totalInSol) } ?: false
         val shouldTryReduceAmount = isAllowedToCorrectAmount && !isSourceSol && !isEnoughSolBalance
+        val hasAlternativeFeePayerTokens = alternativeFeePayerTokens
+            .filterNot {
+                val isSelectedFeePayer = it.tokenSymbol == feePayerToken.tokenSymbol
+                val isNotEnoughBalance = it.totalInLamports.isLessThan(totalNeeded)
+                isSelectedFeePayer || isNotEnoughBalance
+            }
+            .isNotEmpty()
         return when {
             // if there is not enough SPL token balance to cover amount and fee, then try to reduce input amount
             shouldTryReduceAmount && sourceTokenTotal.isLessThan(totalNeeded) -> {
```

Comment:
you should change this as well. It will try to reduce amount even if you have alternative fee payers

The following are priorities: 
1. Source SPL token as a fee payer
2. If source token balance is not enough to pay fees, you should select SOL
3. If source token balance AND SOL is not enough, you should try alternative fee payer
4. If no alternative tokens found, then reduce amount